server/resource_server/service/service.go

```diff
@@ -172,7 +182,46 @@ func (s *service) ProfileClient(userName string) (models.ClientResponseOutput, e
 }
 
 func (s *service) VerifyEmail(userID uint) error {
-	return s.repository.VerifyEmail(userID)
+	user, e := s.repository.FindUser(userID)
+	if e != nil {
+		return e
+	}
+
+	verificationCode := s.emailVerifier.GenerateVerificationCode(&user)
+
+	e = s.emailVerifier.SendVerificationEmail(&user, verificationCode)
+	if e != nil {
+		return e
+	}
+
+	e = s.repository.SaveEmailVerificationData(
+		models.EmailVerificationData{
+			UserId:           user.ID,
+			VerificationCode: verificationCode,
+			ExpiresAt:        time.Now().Add(s.verificationCodeValidityDuration).UTC(),
+		},
+	)
+
+	return e
+}
+
+func (s *service) CheckEmailVerificationCode(userId uint, code string) error {
```

Comment:
This function should be broken into two to follow the single responsibility principle. It both ChecksVerificationCode & then calls `s.repository.SaveProofOfEmailVerification`

Reply:
Line 260 of `controller.go` I feel should be a two step process: 
1) `proof, err := service.CheckEmailVerificationCode(userID, request.Code)`
2) `err := service.SaveProofOfEmailVerification(userID, request.Code, proof)`

Reply:
I broke it to three:

1. ```CheckEmailVerificationCode(userID uint, code string) error```
2. ```GenerateZkProofOfEmailVerification(userID uint) (string, error)```
3. ```SaveProofOfEmailVerification(userID uint, verificationCode string, zkProof string) error```

The second one is to be implemented in the next PR, now it's just a mock.